### PR TITLE
Uncurses Polycircuit Stack Marginally.

### LINKED ISSUE #48420: Polycircuit Aggregate Menu Doesn't Close
## Round ID:
recreate it

## Testmerges: 
none

## Reproduction: 
1. Use polycircuit aggregate
2. Get desired circuit
3. Cry because the menu won't go away
4. Realize that you can summon the polycircuit to you at any time by using the menu that won't close

### PR DESCRIPTION
**Pull request by ArcaneMusic**
## About The Pull Request

Fixes #48420 hopefully. Circuit stack now has a cancel button in the switch menu, as well as a distance check, so that if you just leave it alone it won't start sticking to your arms and teleporting all around the house like a chucky rip off.

## Why It's Good For The Game

Fixes a bug

## Changelog
:cl:
fix: The Polycircuit Stack item no longer leaps through space and time until a circuit is removed.
/:cl:

### CODE REVIEW

File: code/game/objects/items/devices/polycircuit.dm [hunk 1]
```diff
@@ -18,10 +18,15 @@
 	else
```
Comment (kriskog) [inline code/game/objects/items/devices/polycircuit.dm]: Github wont let me comment there, but line 16 change the check to:
if(user.get_inactive_held_item() != src)

File: code/game/objects/items/devices/polycircuit.dm [hunk 2]
```diff
@@ -18,10 +18,15 @@
...
 		if(zero_amount())
 			return
-		chosen_circuit = input("What type of circuit would you like to remove?", "Choose a Circuit Type", chosen_circuit) in list("airlock","firelock","fire alarm","air alarm","APC")
+		chosen_circuit = input("What type of circuit would you like to remove?", "Choose a Circuit Type", chosen_circuit) in list("airlock","firelock","fire alarm","air alarm","APC","cancel")
 		if(zero_amount())
 			return
+		if(get_dist(user, src) > 1)
```
Comment (kriskog) [inline code/game/objects/items/devices/polycircuit.dm]: if(loc != user) or even safer a repeat of the offhand check from earlier.

**Review by kriskog — CHANGES_REQUESTED** [review]
There is another issue where if used correctly in offhand, it will split a single sheet from the stack and put it under your feet as you finish removing your circuit. Likely related to line 39, where you create the new circuit inside the stack item. Try to create it on get_turf(user) instead, it shouldn't show on the floor before put in hand.

File: code/game/objects/items/devices/polycircuit.dm (hunk 1 above)
Comment (kriskog) [reply code/game/objects/items/devices/polycircuit.dm]: Fixes the picking up issue.

**Comment by ArcaneMusic** [comment]
Whoops.

File: code/game/objects/items/devices/polycircuit.dm [hunk 3]
```diff
@@ -13,15 +13,20 @@
...
 	var/mob/living/carbon/human/H = user
-	if(!user.get_inactive_held_item() == src)
+	if(user.get_inactive_held_item() != src)
 		return ..()
 	else
 		if(zero_amount())
 			return
-		chosen_circuit = input("What type of circuit would you like to remove?", "Choose a Circuit Type", chosen_circuit) in list("airlock","firelock","fire alarm","air alarm","APC")
+		if(loc != user)
+			return
```
Comment (kriskog) [inline code/game/objects/items/devices/polycircuit.dm]: You need to check this after the input, since thats when you do stuff.

**Comment by kriskog** [comment]
Did you read the review comment too?

**Pull request merged**